src/org/zaproxy/zap/extension/ascanrulesAlpha/IntegerOverflow.java

```diff
@@ -0,0 +1,269 @@
+			requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR)
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					this.getBaseMsg().getRequestHeader().getURI().toString(), 
+    					param, 
+    					msg.getRequestHeader().toString(), 
+    					this.getError('2') ,
+    					msg);
+    			return;
+    		}
+    		msg = getNewMsg();
+			returnAttack = singleString(4,'1');// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR)
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					this.getBaseMsg().getRequestHeader().getURI().toString(), 
+    					param, 
+    					msg.getRequestHeader().toString(), 
+    					this.getError('3') ,
+    					msg);
+    			return;
+    		}
+    		msg = getNewMsg();
+			returnAttack = singleString(4,'9');// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR)
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					this.getBaseMsg().getRequestHeader().getURI().toString(), 
+    					param, 
+    					msg.getRequestHeader().toString(), 
+    					this.getError('4') ,
+    					msg);
+    			return;
+    		}
+    			
+				return;	
+
+			
+		} catch (Exception e) {
+			log.error(e.getMessage(), e);
+		}	
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		// The CWE id
+		return 190;
+	}
+
+	@Override
+	public int getWascId() {
+		// The WASC ID
+		return 3;
+	}
+	
+	private String randomIntegerString(int length)
+	{
+		
+		int numbercounter = 0;
+		int character = 0;
+		long charactercounter = 0;
+		long maxcharacter = 2147483647;
+		StringBuilder sb1 = new StringBuilder();
+		while (numbercounter < length)
+		{
+			charactercounter = 0;
+			while (charactercounter < maxcharacter)

```

Comment:
This loop doesn't look right, it tries to create a string with 2147483647 characters (and if we include the previous loop it would result in 2147483647 * ```length``` characters). 
Running the scanner fails with ```OutOfMemoryError: Java heap space``` and if it has enough heap space it fails with ```OutOfMemoryError: Requested array size exceeds VM limit```.